### PR TITLE
RealNVP

### PR DESCRIPTION
**Pull request by stefanwebb**
The affine coupling transform from RealNVP

### CODE REVIEW

**Comment by stefanwebb** [comment]
@martinjankowiak this is ready to be looked at

File: pyro/distributions/transforms/affine_coupling.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.distributions import constraints
+
+from pyro.distributions.torch_transform import TransformModule
+from pyro.distributions.util import copy_docs_from
+
+
+def clamp_preserve_gradients(x, min, max):
```
Comment (martinjankowiak) [inline pyro/distributions/transforms/affine_coupling.py]: if this is being reused in several places it should be put in a utils file somewhere

Comment (stefanwebb) [reply pyro/distributions/transforms/affine_coupling.py]: Done!

**Comment by martinjankowiak** [comment]
i guess if you rebase off of #1933 tests will pass? or do you want to certify that the failing test is definitively spurious?

**Comment by stefanwebb** [comment]
I think it's probably not worth checking since it passes on my machine and we can clear the backlog by merging this one in

**Pull request merged**